skills_ml/algorithms/job_sampler/sampler.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import heapq as hq
+import random
+
+def reservoir(it, k):
+    """
+    Reservoir sampling with Random Sort from a job posting iterator.
+
+    Randomly choosing a sample of k items from a streaming iterator. Using random sort to implement the algorithm.
+    Basically, it's assigning random number as keys to each item and maintain k items with minimum value for keys,
+    which equals to assigning a random number to each item as key and sort items using these keys and take top k items.
+
+    Args:
+        it (iterator): Job posting iterator to sample from
+        k (int): Sample size
+
+    Returns:
+        list: The result sample of k items.
+    """
+    it = iter(it)
+    result = []
+    for i, datum in enumerate(it):
+        if i < k:
+            result.append(datum)
+        else:
+            j = random.randint(0, i-1)
+            if j < k:
+                result[j] = datum
+    return result
+
+
+def reservoir_weighted(it, k, weights):
```

Comment:
Both `reservoir` and `reservoir_weighted` seem pretty generic. Should they just go into an `algorithms.sampling` module? 

This job stuff could even be in a submodule `algorithms.sampling.job`

For that matter, we have a generic `file_sampler` which has similar logic to the unweighted version. Should that be merged with this?